app/controllers/application_controller.rb

```diff
@@ -27,6 +24,34 @@ def admin_required
     redirect_to root_path unless current_user && current_user.admin?
   end
 
+  # This method is used to verify if the user is already logged in,
+  # and will redirect it to the homepage in case it is not.
+  # This method overrides a default behavior from Devise to allow auth0
+  # authentication working simultaneously with legacy devise database mode,
+  # and should be refactored once we fully migrate to auth0.
+  def authenticate_user!
+    resp = check_auth0_auth_token
+    if resp.nil?
+      # invoke devise if auth0 token is not present
+      super
+    else
+      if resp[:authenticated]
+        auth_payload = resp[:auth_payload]
+        return if current_user&.email == auth_payload["email"]
+        auth_user = User.find_by(email: auth_payload["email"])
+        if auth_user.present?
+          sign_in auth_user, store: false
```

Comment:
what does `store: false` do?

Reply:
Devise documentation doesn't have info about that, but from what I could check, it prevents warden from creating a session, which is exactly what I wanted since I'm handling the session in a different place.
I could find some extra info about this behavior [here](https://github.com/plataformatec/devise/issues/3981)